packages/headless/src/controllers/case-input/headless-case-input.ts

```diff
@@ -0,0 +1,152 @@
+import {logUpdateCaseField} from '../../features/case-assist/case-assist-analytics-actions';
+import {CaseAssistEngine} from '../../app/case-assist-engine/case-assist-engine';
+import {
+  CaseAssistConfigurationSection,
+  CaseFieldSection,
+  CaseInputSection,
+  ConfigurationSection,
+  DocumentSuggestionSection,
+} from '../../state/state-sections';
+import {buildController, Controller} from '../controller/headless-controller';
+import {
+  caseAssistConfiguration,
+  caseField,
+  caseInput,
+  configuration,
+  documentSuggestion,
+} from '../../app/reducers';
+import {Schema, StringValue} from '@coveo/bueno';
+import {validateOptions} from '../../utils/validate-payload';
+import {loadReducerError} from '../../utils/errors';
+import {fetchDocumentSuggestions} from '../../features/document-suggestion/document-suggestion-actions';
+
+export interface CaseInputOptions {
+  field: string;
+}
+
+export interface CaseInputProps {
+  options: CaseInputOptions;
+}
+
+function validateCaseInputOptions(
+  engine: CaseAssistEngine,
+  options: Partial<CaseInputOptions> | undefined
+) {
+  const schema = new Schema<CaseInputOptions>({
+    field: new StringValue({
+      required: true,
+      emptyAllowed: false,
+    }),
+  });
+  validateOptions(engine, schema, options, 'buildCaseInput');
+}
+
+export interface UpdateFetchOptions {
+  caseClassifications?: boolean;
+  documentSuggestions?: boolean;
+}
+
+/**
+ * The `CaseInput` controller is responsible for setting and retrieving the value of a single field from the case creation form and optionally trigger Case Assist API requests.
+ */
+export interface CaseInput extends Controller {
+  /**
+   * Sets the value of the specified field.
+   *
+   * @param value - The case input value to set.
+   * @param updatesToFetch - A set of flags dictating whether to fetch case assist data after updating the input value.
+   */
+  update(value: string, updatesToFetch?: UpdateFetchOptions): void;
+  /**
+   * A scoped and simplified part of the headless state that is relevant to the `CaseInput` controller.
+   */
+  state: CaseInputState;
+}
+
+export interface CaseInputState {
+  /**
+   * The value of the case input.
+   */
+  value: string;
+}
+
+/**
+ * Creates a `Case Input` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `CaseInput` controller properties.
+ * @returns A `CaseInput` controller instance.
+ */
+export function buildCaseInput(
+  engine: CaseAssistEngine,
+  props: CaseInputProps
+): CaseInput {
+  if (!loadCaseInputReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildController(engine);
+  const {dispatch} = engine;
+  const getState = () => engine.state;
+
+  validateCaseInputOptions(engine, props.options);
+
+  const fieldName = props.options.field;
+
+  dispatch(
+    updateCaseInput({
+      fieldName: fieldName,
+      fieldValue: '',
```

Comment:
Does it make sense to support an `initialFieldValue` here ? That would get passed in through `CaseInputProps` ?

Reply:
I can't really think of a scenario where you would want to instantiate your case input with a pre-filled initial value. And if you want to be weird then you can simply call `update()` with the value right after creating the engine to set it. So no I don't think it's necessary.

Reply:
The only reason I could see is if you allow to edit an existing case. Then it would make sense to be able to initialize the case input with a specific value. Calling `update()` would record analytics actions while the user didn't perform any real action.

Reply:
That's interesting. Do you know if that something that companies implement?

Reply:
> Do you know if that something that companies implement?

I've not seen it implemented that way so far. So I'd suggest to remove `initialFieldValue` and add it if needed.